mu/modes/microbit.py

```diff
@@ -275,9 +287,18 @@ def flash(self):
                 information = _("Your script is too long!")
                 self.view.show_message(message, information, 'Warning')
                 return
+        # By this point, there's a valid Python script in "python_script".
+        # Assign this to an attribute for later processing in a different
+        # method.
+        self.python_script = python_script
+        # Next step: find the microbit port and serial number.
+        path_to_microbit = uflash.find_microbit()
+        port, serial_number = microfs.find_microbit()
```

Comment:
Right... but this is *micro:bit* mode. If you want to use a *sino:bit* or calliope or some other device, then either write a new mode that uses the serial number (it's available) or simply don't do that! ;-)

Reply:
To be honest, it's quite possible the sino:bit and calliope are both using the same VID/PID and Board ID, as those are defined in DAPLink (open source project), they are using the same DAPLink KL26, I haven't found any forks on their GH orgs, and I don't think Mbed has issued new IDs for those boards.
So even with those checks in place this mode would probably still work on those boards.

Reply:
Right... so if we can't distinguish between boards, then it's a user/upstream project's problem.

Mu only advertises micro:bit mode will support "real" micro:bits. If Calliope / sino:bit want to use their own build of MicroPython (which they should, since they have different hardware requirements) then it's VERY easy to create a new mode in Mu.

We've done all we can in this case. I don't see why Mu should do more work for people who try to use a mode with the wrong device.

Reply:
A Mbed board that can run micro:bit MicroPython is fine if it flashes in the micro:bit mode and that's not really a problem. If it works great, if it doesn't it's probably to be expected. This is not an issue or concern.

An Mbed board that **cannot** run micro:bit MicroPython can give you a false positive here and be identified as a micro:bit. To avoid that the Product ID can be checked.

Reply:
Or simply don't plug that sort of board into Mu and expect it to work. If you're going to shoot yourself in the foot, Mu shouldn't try to stop you.